metricbeat/tests/system/test_docker.py

```diff
@@ -11,17 +12,33 @@ def test_container_fields(self):
         """
         test container fields
         """
-        self.render_config_template(modules=[{
-            "name": "docker",
-            "metricsets": ["container"],
-            "hosts": ["unix:///var/run/docker.sock"],
-            "period": "10s",
-        }])
+
+        test_env = os.environ.get('DOCKER_COMPOSE_PROJECT_NAME')
+        processors = None
+        if test_env:
+            processors = [{
+                "drop_event": {
+                    "when.not": "contains.docker.container.name: " + test_env,
+                },
+            }]
+
+        self.render_config_template(
+            modules=[{
+                "name": "docker",
+                "metricsets": ["container"],
+                "hosts": ["unix:///var/run/docker.sock"],
+                "period": "10s",
+            }],
+            processors=processors,
+        )
 
         proc = self.start_beat()
         self.wait_until(lambda: self.output_lines() > 0, max_timeout=20)
         proc.check_kill_and_wait()
-        self.assert_no_logged_warnings()
+        self.assert_no_logged_warnings(["WARN Container stopped when recovering stats",
+                                        "ERR An error occurred while getting docker stats"])
+
+        print(os.environ.get('TESTING_ENVIRONMENT'))
```

Comment:
I wanted to keep it in for the moment to see what went wrong in case the tests fail again.